web/controllers/game_controller.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Yggdrasil.GameController do
+  use Yggdrasil.Web, :controller
+
+  alias Yggdrasil.Game
+
+  def index(conn, _params) do
+    games = Repo.all Game
```

Comment:
When I have setup a similar layer in the past it has ended up mostly being a pass through layer that replicated what the web needed to do. It didn't add a lot of a value and created a extra work.

I get that we may need to do a bunch of extra work on a character before we remove it, but it might be context specific. Example is that an admin comes in and removes a character, it should auto logout the player if they are playing, send them an email with the reason for removal, and delete the character. If a player came in and tried to delete themselves while they were player we most likely want to return back an error telling the player that they need to logout first as basically a means to ensure they didn't do it accidentally while trying to view something about their account while player. The game engine won't know these things and we could end up with a large API with little value.

I think it is simpler to start off without the layer of abstraction and move towards it if we start feeling the pain when making changes. Maintainable code is code that you can change easily, not just how things are organized. Sort of the YAGNI and evolutionary design approach. We have the luxury that there is not deadline and we can make these refactorings as we feel them.

I feel the same thing about splitting things out into separate apps too early. It adds concerns about ensuring we don't run into circular dependencies or confusing and leaky APIs because we made a bad assumption that something could be split out.